src/interface/data_collection.hpp

```diff
@@ -42,21 +44,36 @@ class DataCollection {
 
   void SetMeshPointer(Mesh *pmesh) { pmy_mesh_ = pmesh; }
 
-  std::shared_ptr<T> &Add(const std::string &label, const std::shared_ptr<T> &src,
-                          const std::vector<std::string> &flags, const bool shallow);
-  std::shared_ptr<T> &Add(const std::string &label, const std::shared_ptr<T> &src,
-                          const std::vector<std::string> &flags = {});
-  std::shared_ptr<T> &AddShallow(const std::string &label, const std::shared_ptr<T> &src,
-                                 const std::vector<std::string> &flags = {});
-  std::shared_ptr<T> &Add(const std::string &label) {
-    // error check for duplicate names
-    auto it = containers_.find(label);
+  template <typename ID_t>
+  std::shared_ptr<T> &Add(const std::string &name, const std::shared_ptr<T> &src,
+                          const std::vector<ID_t> &fields, const bool shallow) {
+    auto it = containers_.find(name);
     if (it != containers_.end()) {
+      if (!(it->second)->Contains(fields)) {
+        PARTHENON_THROW(name +
+                        " already exists in collection but does not contain fields");
+      }
```

Comment:
Does this check that `fields` is exactly the same set of fields as are contained in `it->second` or just that the requested fields are a subset of the fields in `it->second`? I am not sure which is the desired behavior, but I would lean towards the former.

Reply:
It's the latter. This checks that the requested fields are all available in `it->second`. And I would suggest that's desired behavior.

Reply:
I guess the concern I have is for communication. If you mistakenly use the same name for two mesh_data, one of which contains a small subset of the fields of the other you will possibly end up communicating extra fields you didn't intend.

Reply:
Hmm... That's true. I can change this, but that would be a change in functionality, as I just preserved the existing functionality (probably created by @jdolence ?)

Reply:
I would just leave it as is.

Reply:
Too late fixed it. ;)